operations/configuration_management/Management_Node_Image_Customization.md

```diff
@@ -0,0 +1,163 @@
+   ```json
+   "layers": [
+   {
+     "name": "csm-ncn-workers",
+     "cloneUrl": "https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git",
+     "playbook": "ncn-worker_nodes.yml",
+     "commit": "<git commit hash>"
+   },
+   ```
+
+  The last layer in the CFS session should be similar to this:
+
+   ```json
+   "layers": [
+   {
+     "name": "csm-ncn-initrd",
+     "cloneUrl": "https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git",
+     "playbook": "ncn-initrd.yml",
+     "commit": "<git commit hash>"
+   }
+   ```
+
+1. (`ncn-mw#`) Update NCN boot parameters.
+
+    1. Get the existing `metal.server` setting for the component name (xname) of the node of interest:
+
+        ```bash
+        XNAME=<node-xname>
+        METAL_SERVER=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' \
+            | awk -F 'metal.server=' '{print $2}' \
+            | awk -F ' ' '{print $1}')
+        echo "${METAL_SERVER}"
+        ```
+
+    1. Update the kernel, `initrd`, and metal server to point to the new artifacts.
+
+        **NOTE:** `${IMS_RESULTANT_IMAGE_ID}` is the `result_id` returned in the output of the last command
+        in the "Create an Image Customization CFS Session" procedure, repeated here for convenience:
+
+        ```bash
+        cray cfs sessions describe example --format json | jq .status.artifacts
+        ```
+
+        ```bash
+        S3_ARTIFACT_PATH="boot-images/${IMS_RESULTANT_IMAGE_ID}"
+        NEW_METAL_SERVER="s3://${S3_ARTIFACT_PATH}/rootfs"
+
+        PARAMS=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' | \
+            sed "/metal.server/ s|${METAL_SERVER}|${NEW_METAL_SERVER}|" | \
+            sed "s/metal.no-wipe=1/metal.no-wipe=0/" | \
+            tr -d \")
+        echo "${PARAMS}"
+        ```
+
+        In the output of the final echo command, verify that the value of `${NEW_METAL_SERVER}` was set correctly.
+
+    1. Update BSS with the new boot parameters.
+
+        ```bash
+        cray bss bootparameters update --hosts "${XNAME}" \
+            --kernel "s3://${S3_ARTIFACT_PATH}/kernel" \
+            --initrd "s3://${S3_ARTIFACT_PATH}/initrd" \
+            --params "${PARAMS}"
+        ```
+
+1. (`ncn-mw#`) Prepare for reboot.
+
+   **NOTE**: If the worker node image is being customized as part of a Cray EX initial install or upgrade involving multiple products,
+   then refer to the /HPE Cray EX System Software Getting Started Guide/ (S-8000) for details on when to reboot the worker nodes to the new image.
+
+   1. On the nodes being rebooted, run the following command to disable the boot loader and prepare the node to accept a new SquashFS.
+
+      ```bash
+      rm -rvf /metal/recovery/*
+      ```
+
+   1. Failover any Postgres leader that is running on the worker node being rebooted.
+
+      ```bash
+      /usr/share/doc/csm/upgrade/1.2/scripts/k8s/failover-leader.sh <node to be rebooted>
```

Comment:
Should `<node to be rebooted>` be an xname? That could be clarified, IMO.

Reply:
An xname may work, I'm not sure.  But `ncn-wXXX` does work.  I'll clarify.

Reply:
After talking with Dean, I removed this section.